src/main/java/org/openlmis/buq/service/buq/BottomUpQuantificationService.java

```diff
@@ -288,6 +295,46 @@ public void delete(BottomUpQuantification bottomUpQuantification) {
     bottomUpQuantificationRepository.deleteById(bottomUpQuantification.getId());
   }
 
+  /**
+   * Calculates and retrieves statistics related to the approval of facility forecasting for a
+   * given program.
+   *
+   * @param programId The UUID of the program for which statistics are calculated.
+   * @return {@link ApproveFacilityForecastingStats} containing the calculated data.
+   */
+  public ApproveFacilityForecastingStats getApproveFacilityForecastingStats(UUID programId) {
+    UserDto currentUser = authenticationHelper.getCurrentUser();
+    List<String> userPermissionStrings = userReferenceDataService
+        .getPermissionStrings(currentUser.getId());
+
+    List<UUID> userSupervisedFacilities = userPermissionStrings.stream()
+        .filter(p -> p.length() - p.replace("|", "").length() == 2
+            && APPROVE_BUQ_RIGHT_NAME.equals(p.substring(0, p.indexOf('|'))))
+        .map(p -> p.split("\\|"))
+        .filter(p -> programId.equals(UUID.fromString(p[2])))
+        .map(p -> UUID.fromString(p[1]))
+        .collect(Collectors.toList());
+    int totalFacilities = userSupervisedFacilities.size();
+
+    List<BottomUpQuantification> bottomUpQuantifications =
+        bottomUpQuantificationRepository.findByFacilityIdIn(userSupervisedFacilities);
+    int totalBottomUpQuantifications = bottomUpQuantifications.size();
+
+    if (totalBottomUpQuantifications == 0) {
+      return new ApproveFacilityForecastingStats(totalFacilities, 0, 0);
+    } else {
```

Comment:
nitpicky: `else` is not required if you return from the statement above.

Reply:
I removed "else" statement.